### ISSUE TITLE
intelliSense broken

### ISSUE THREAD

**Issue by SmileGobo** [microsoft/vscode-cpptools#8887]
Bug type: General

**vscode**:
Version: 1.64.2 (user setup)
Commit: f80445acd5a3dadef24aa209168452a3d97cc326
Date: 2022-02-09T22:02:28.252Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 6.1.7601
C/C++ Extension Version: v1.8.4 (1.8.2)
cmake tools:  v1.9.2

1. Open cmake project folder
2. open .cpp or .h file

**Expected behavior**

intelliSense not work, autocomplete not work, navigate not work

**Additional context**

Error on extension console
```
[2022-02-19 15:04:21.334] [exthost] [warning] TextEditor is closed/disposed
[2022-02-19 15:04:55.678] [exthost] [error] [ms-vscode.cpptools] provider FAILED
[2022-02-19 15:04:55.678] [exthost] [error] TypeError: Cannot read property 'ActiveClient' of undefined
	at Object.<anonymous> (c:\Users\user\.vscode\extensions\ms-vscode.cpptools-1.8.4\dist\main.js:39247:37)
	at Generator.next (<anonymous>)
	at c:\Users\user\.vscode\extensions\ms-vscode.cpptools-1.8.4\dist\main.js:39062:71
	at new Promise (<anonymous>)
```

**Comment by BillDenton** [comment]
I appear to have the same or similar issue. None of the C/C++ extension commands work. For example:
![image](https://user-images.githubusercontent.com/17859340/154979051-f8d1d461-0bfb-4898-9283-66fa946610f4.png)
In the Log (Extension Host):
[2022-02-21 14:51:10.277] [exthost] [error] [ms-vscode.cpptools] provider FAILED
[2022-02-21 14:51:10.277] [exthost] [error] TypeError: Cannot read property 'ActiveClient' of undefined

The whole extension is completely unusable. Insider version:
Name: C/C++
Id: ms-vscode.cpptools
Description: C/C++ IntelliSense, debugging, and code browsing.
Version: 1.9.0
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools

**Comment by BillDenton** [comment]
Tried uninstalling and re-installing. Still failing in the same way. So the C/C++ extension is completely broken as it doesn't load.

**Comment by Colengms** [comment]
Hi @SmileGobo @BillDenton .  Could look in the `.vscode\extensions\ms-vscode.cpptools-x.x.x\.vsixmanifest` in your profile directory, and tell me what the value of `TargetPlatform` there is?  I'm curious if you're repro'ing this issue: https://github.com/microsoft/vscode/issues/141696

**Comment by Colengms** [comment]
I'm not certain this issue has been addressed, but there will be a couple of fixes in 1.9.1 that might address this.  1.9.1 should be available 'really soon'.

**Comment by BillDenton** [comment]
@colengms In my .vscode\extensions\ms-vscode.cpptools-1.9.0 in .vsixmanifest:
<Identity Language="en-US" Id="cpptools" Version="1.9.0" Publisher="ms-vscode" TargetPlatform="win32-x64"/>

Checking in the Log (Extension Host) today I also noticed another error from a different extension:
[2022-02-22 22:10:14.131] [exthost] [info] ExtensionService#_doActivateExtension beaglefoot.awk-ide-vscode, startup: false, activationEvent: 'onLanguage:awk'
[2022-02-22 22:10:21.525] [exthost] [error] [beaglefoot.awk-ide-vscode] provider FAILED
[2022-02-22 22:10:21.525] [exthost] [error] Error: Request getSemanticTokens failed with message: Cannot read property 'getLanguage' of undefined

....

	at handleResponse (c:\Users\den71958\.vscode\extensions\beaglefoot.awk-ide-vscode-0.9.0\node_modules\vscode-jsonrpc\lib\common\connection.js:477:48)
	at processMessageQueue (c:\Users\den71958\.vscode\extensions\beaglefoot.awk-ide-vscode-0.9.0\node_modules\vscode-jsonrpc\lib\common\connection.js:292:17)
	at Immediate.<anonymous> (c:\Users\den71958\.vscode\extensions\beaglefoot.awk-ide-vscode-0.9.0\node_modules\vscode-jsonrpc\lib\common\connection.js:276:13)
	at processImmediate (internal/timers.js:461:21)
[2022-02-23 09:05:53.340] [exthost] [error] [ms-vscode.cpptools] provider FAILED
[2022-02-23 09:05:53.340] [exthost] [error] TypeError: Cannot read property 'ActiveClient' of undefined

I recently installed the AWK extension (beaglefoot.awk-ide-vscode). Could that be causing the problem? I'll try uninstalling it.

**Comment by BillDenton** [comment]
@bobbrow @colengms any updates? The extension is unusable. I can't switch to the Clangd extension as our build system (scons based) doesn't produce the compiler json database. Tnanks.

**Comment by Colengms** [comment]
Hi @BillDenton @SmileGobo .  1.9.1 has been released.  Could you check if this issue is still reproducible with 1.9.1?

**Comment by xjrjyy** [comment]
In my VS Code, cpptools can't use it with clangd(because of the Intelli Sense Engine setting)

` [error] Error: Do not activate the extension when IntelliSense is disabled.`

**Comment by BillDenton** [comment]
@xjrjyy As far as I can tell the Microsoft C/C++ extension is mutually exclusive with clangd extension. As the Microsoft extension wasn't starting I tried the clangd extension (after disabling the Microsoft extension).
@Colengms after upgrading to 1.9.1 is now working again. However, it did start working with just a window reload. I had to shut down VS Code and start again. I found that the IntelliSense had been disabled (like @xjrjyy). After re-enabling and restarting VS Code is now working. Glad to have it back again. It was quite a pain to navigate the code without it. Thaks.
